server/docs/swagger/report.yaml

```diff
@@ -0,0 +1,141 @@
+paths:
+  /articles/{slug}/reports/:
+    post:
+      tags:
+        - Report
+      summary: create a new article report
+      produces:
+      - application/json
+      parameters:
+      - name: authorization
+        in: header
+        description: authorization token
+        type: string
+      - name: slug
+        in: path
+        description: path parameter for the article slug
+        required: true
+        type: string
+      - in: body
+        name: report create
+        description: It enables a user to report an article
+        required: true
+        schema:
+          $ref: '#/definitions/reportArticle'
+      responses:
+        201:
+          description: An object with report details
+        400:
+          description: Invalid query parameters
+        404:
+          description: Resource not found
+  /reports/:
+    get:
+      tags:
+        - Report
+      summary: gets all the reports
+      produces:
+      - application/json
+      parameters:
+      - name: authorization
+        in: header
+        description: authorization token
+        type: string
+      - in: query
+        name: limit
+        description: Set the limit for returned reports
+      - in: query
+        name: page
+        description: Set the page for returned reports
+      - in: query
+        name: order
+        description: Set the order for returned reports
+        type: string
+      - name: slug
+        in: path
+        description: Set the Article slug
+      responses:
+        200:
+          description: Success response
+        400:
+          description: Invalid query parameters
+  /reports/{reportId}:
+    get:
+      tags:
+        - Report
+      summary: gets all the reports
+      produces:
+      - application/json
+      parameters:
+      - name: authorization
+        in: header
+        description: authorization token
+        type: string
+      - name: reportId
+        in: path
+        description: path parameter takes the report id
+        required: false
+        type: string
+      responses:
+        200:
+          description: Success response
+        400:
+          description: Invalid query parameters
+        404:
+          description: Resource not found
+    put:
+      tags:
+        - Report
+      summary: It updates a user's report on an article
+      produces:
+      - application/json
+      parameters:
+      - name: authorization
+        in: header
+        description: authorization token
+        type: string
+      - name: reportId
+        in: path
+        description: path parameter takes the report id
+        required: true
```

Comment:
remove the `required: true` to allow error message from the server sent to swagger